Source/Shaders/Appearances/PolylineMaterialAppearanceVS.glsl

```diff
@@ -10,20 +10,45 @@ attribute float batchId;
 
 varying float v_width;
 varying vec2 v_st;
+varying float v_angle;
 
-void main() 
+void main()
 {
     float expandDir = expandAndWidth.x;
     float width = abs(expandAndWidth.y) + 0.5;
     bool usePrev = expandAndWidth.y < 0.0;
-    
+
     vec4 p = czm_computePosition();
     vec4 prev = czm_computePrevPosition();
     vec4 next = czm_computeNextPosition();
-    
+
+    // Compute the points in eye coordinates.
+    vec4 prevEC = czm_modelViewRelativeToEye * prev;
+    vec4 nextEC = czm_modelViewRelativeToEye * next;
+    vec4 pEC = czm_modelViewRelativeToEye * p;
+
+    // Compute the positions in clip space.
+
+    vec4 prevClip = czm_viewportOrthographic * prevEC;
+    vec4 nextClip = czm_viewportOrthographic * nextEC;
+    vec4 pClip = czm_viewportOrthographic * pEC;
+
+    // Determine the relative screen space direction of the line.
+    vec2 dir;
+    if (usePrev) {
+        dir = normalize(pClip.xy - prevClip.xy);
+    }
+    else {
+        dir = normalize(nextClip.xy - pClip.xy);
+    }
+    v_angle = atan(dir.x, dir.y) - atan(1.0, 0.0);
```

Comment:
OK, I found a 168 meg CZML file full of polylines, that's quite a stress test bringing Cesium down to about 5fps on my Quadro K2000M.  I tried this with the performance counter turned on, and it appears to hover around 170ms per frame with or without this branch.  So, for my card at least, there's no performance concern.